### PR TITLE
TLMBLGIS-50 - use grunt to validate and concatenate js files

### PR DESCRIPTION
**Pull request by gsilver**
for
https://itsjira.umms.med.umich.edu/browse/TLMBLGIS-50

### CODE REVIEW

**Comment by lsloan** [comment]
Since I've not used grunt before, I'd like to talk to you about this soon to make sure I understand what this is doing.

**Comment by lsloan** [comment]
Q: The JIRA issue says that grunt is to be run every time the VM is brought up.  However, it looks to me like it's only being run at provisioning time.  That is, when the VM is created the first time or anytime "vagrant provision" is run.  It's not being run with every "vagrant up".

On the other hand, I don't think running it every time the VM comes up is the right time.   It doesn't seem like it would be often enough.   When I'm developing with a VM running, I don't stop and restart the VM with each code change.  If you want this to be more automatic, something else needs to be done, right?

**Comment by gsilver** [comment]
@lsloan - sure let's talk about it - let me know when is good

File: Vagrantfile [hunk 1]
```diff
@@ -59,6 +59,10 @@ EOM
         apt-get install --yes nodejs
         npm install --global npm@latest
         npm install --global bower
-
+        npm install -g grunt-cli
+        npm install --global grunt
+        cd mbofui
+        npm install
```
Comment (lsloan) [inline Vagrantfile]: We talked about this a little this afternoon.  Looking more closely, I think this line, `npm install`, is unnecessary.  It's already run on line 60, above.

**Comment by lsloan** [comment]
As we discussed this afternoon, the `grunt` command in `Vagrantfile` will only be run when the VM is installed/reinstalled, not whenever it starts.  Two options for where it should go:
1. In `~vagrant/.bashrc` or some similar file specific to the vagrant user.  I think it will run only when the vagrant user is logged in, like when the developer runs `vagrant ssh`.  I could be wrong, but I think it will only run then.
2. In some system startup file, like /etc/init.d/something...  I'm not sure where is appropriate for this version of Linux.

That said, I'm not sure what the `grunt` command is supposed to do.  I thought it would just run the tasks from `Gruntfile.js` and exit.  Does loading `grunt-contrib-watch` make it run in the background waiting for those watched files to change?

PS:  Have you tried reinstalling the VM and see how your changes work?

File: mbofui/app/index.html [hunk 2]
```diff
@@ -3,14 +3,14 @@
 
 <head>
   <meta charset="utf-8">
-  <title>{{strings.app_name}}
+  <title>Ocellus
```
Comment (dovek) [inline mbofui/app/index.html:6]: Is this something we want to hardcode? I thought we were going to get this from a properties file if we changed the name or something.

File: mbofui/app/js/app.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-'use strict';
+/*jshint strict:false */
 /* global angular */
```
Comment (dovek) [inline mbofui/app/js/app.js:2]: Does 'use strict' go away because we are using grunt?

File: mbofui/package.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+
+{
+  "name": "Ocellus",
```
Comment (dovek) [inline mbofui/package.json:3]: As mentioned above, I thought we were keeping the name of the application here. Why is it hard coded in index.html?

File: mbofui/app/index.html (hunk 2 above)
Comment (gsilver) [inline mbofui/app/index.html:6]: Yeah - I noticed that when you bookmark it it uses the {{some_variable}} thing.

File: mbofui/app/js/app.js (hunk 3 above)
Comment (gsilver) [inline mbofui/app/js/app.js:2]: Yeah - strict mode was too bossy.

**Comment by gsilver** [comment]
@lsloan  - fixed it via:

```
config.vm.provision "shell", inline: <<-SHELL
# install stuff here
SHELL
# added - will always run (vagrant up and vagrant reload)
config.vm.provision "shell", run: 'always',inline: <<-SHELL2
cd /vagrant/mbofui
npm install
grunt --gruntfile Gruntfile-vagrant.js
SHELL2

```

**Comment by lsloan** [comment]
Nice to know about the `run: 'always'` option for Vagrant's provisioning.

File: Vagrantfile [hunk 5]
```diff
@@ -59,6 +59,12 @@ EOM
...
         npm install --global bower
-
+        npm install -g grunt-cli
+        npm install --global grunt
     SHELL
+    config.vm.provision "shell", run: 'always',inline: <<-SHELL2
+    cd /vagrant/mbofui
+    npm install
+    grunt --gruntfile Gruntfile-vagrant.js
+    SHELL2
```
Comment (lsloan) [inline Vagrantfile:69]: Indent the lines between the `config.vm.provision` line and this closing `SHELL2` line.  That will improve readability and show more clearly what's being done in this provisioning step.

**Pull request merged**